cmake/modules/boards.cmake

```diff
@@ -128,6 +128,9 @@ if(DEFINED BOARD_REVISION)
     set(BOARD_REVISION ${ACTIVE_BOARD_REVISION})
   endif()
 
+  # Add board revision to build variables
+  add_definitions(-DCONFIG_BOARD_REVISION="${BOARD_REVISION}")
```

Comment:
Maybe an approach similar to what's being done in `version.h.in` would be cleaner? Maybe even use that very file? See here: https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/gen_version_h.cmake#L26